### ISSUE TITLE
Client always sends extraneous last request

### ISSUE THREAD

**Issue by dcolligan** [ga4gh/server#267]
Since we always return a pageToken regardless if we are at the end of the object stream, the client sees a non-null nextPageToken and decides to send one last request, which always returns an empty payload from the server.

Not sure if there is a way to fix this generally (or if it is even worth it in the specific case, since we would need to do some kind of look-ahead logic with iterators), but it's worth discussing.

```
$ python client_dev.py -vv -O variants-search http://localhost:8000/v0.5.1 -V 1000g_2013.wt -r 1 -c HG03279
... previous requests ...
ga4gh.client: POST http://localhost:8000/v0.5.1/variants/search
ga4gh.client: json request:
ga4gh.client: {
    "callSetIds": [
        "HG03279"
    ],
    "end": 4294967294,
    "pageSize": 100,
    "pageToken": "0:138397",
    "referenceName": "1",
    "start": 0,
    "variantName": null,
    "variantSetIds": [
        "1000g_2013.wt"
    ]
}
ga4gh.client: json response:
ga4gh.client: {
    "nextPageToken": null,
    "variants": []
}
```

**Comment by jeromekelleher** [comment]
This is a bug in the server --- we shouldn't send back a non-null pageToken unless there are records left.

**Comment by dcolligan** [comment]
It looks like `variantsGenerator` and `callSetsGenerator` are the only methods affected by this issue.

**Comment by jeromekelleher** [comment]
We should fix `variantsGenerator` to use the same algorithm as reads (hopefully reusing some code too), as the same complications due to interval search can occur. `callSetsGenerator` will need to be fixed in isolation.

**Comment by jeromekelleher** [comment]
Fixed in #295 and #304.
